crates/iota-indexer/src/models/objects.rs

```diff
@@ -160,6 +180,94 @@ pub struct StoredHistoryObject {
     pub df_kind: Option<i16>,
 }
 
+impl StoredHistoryObject {
+    pub async fn try_into_past_object_read(
+        &self,
+        package_resolver: Arc<Resolver<impl PackageStore>>,
+    ) -> Result<PastObjectRead, IndexerError> {
+        let object_ref = self.get_object_ref()?;
+
+        let object: Object = self.clone().try_into()?;
```

Comment:
I think this can simplify to:
```suggestion
        self,
        package_resolver: Arc<Resolver<impl PackageStore>>,
    ) -> Result<PastObjectRead, IndexerError> {
        let object: Object = self.try_into()?;
        let object_ref = object.compute_object_reference();
```

so that `get_object_ref` can be removed.

Reply:
Nice one, refactored in: https://github.com/iotaledger/iota/pull/5865/commits/afc5550e42ec0a1e6b3ee270ac7875d6b588b5e4